src/connectors/notify/notify.connector.js

```diff
@@ -86,6 +106,14 @@ const sendSingleEmail = async (
       "sendSingleEmail",
       newError
     );
+    if (updateCache) {
```

Comment:
Wouldn't it make sense to just initialise with "completed: false" and skip this step on failure?